pebbles/drivers/provisioning/docker_driver.py

```diff
@@ -134,47 +136,47 @@ def get_pb_client(token, api_base_url, ssl_verify):
         return PBClient(token, api_base_url, ssl_verify)
 
     @classmethod
-    def load_records(cls, token, url):
+    def load_records(cls, token, url, namespace_value=NAMESPACE_CPU):
         """ Loads the pool vm host state from the database through REST API
             There can be multiple pool vms at a time (in different states)
         """
         pbclient = cls.get_pb_client(token, url, ssl_verify=False)
-        namespaced_records = pbclient.get_namespaced_keyvalues({'namespace': NAMESPACE, 'key': KEY_PREFIX_POOL})
+        namespaced_records = pbclient.get_namespaced_keyvalues({'namespace': namespace_value, 'key': KEY_PREFIX_POOL})
         hosts = []
         for ns_record in namespaced_records:
             hosts.append(ns_record['value'])
         return hosts
 
     @classmethod
-    def save_records(cls, token, url, hosts):
+    def save_records(cls, token, url, hosts, namespace_value=NAMESPACE_CPU):
         """Saves the pool vm host state in the database via REST API
         """
         pbclient = cls.get_pb_client(token, url, ssl_verify=False)
         for host in hosts:
             _key = '%s_%s' % (KEY_PREFIX_POOL, host['id'])
             payload = {
-                'namespace': NAMESPACE,
+                'namespace': namespace_value,
                 'key': _key,
                 'schema': {}
             }
             if host.get('state') in [DD_STATE_SPAWNED, DD_STATE_ACTIVE, DD_STATE_INACTIVE]:  # POST or PUT
                 payload['value'] = host
-                pbclient.create_or_modify_namespaced_keyvalue(NAMESPACE, _key, payload)
+                pbclient.create_or_modify_namespaced_keyvalue(namespace_value, _key, payload)
             elif host.get('state') == DD_STATE_REMOVED:  # DELETE
-                pbclient.delete_namespaced_keyvalue(NAMESPACE, _key)
+                pbclient.delete_namespaced_keyvalue(namespace_value, _key)
 
     @classmethod
-    def load_driver_config(cls, token, url):
+    def load_driver_config(cls, token, url, namespace_value=NAMESPACE_CPU):
         """ Loads the driver config from the database through REST API
             There will always be one config for the driver
         """
         pbclient = cls.get_pb_client(token, url, ssl_verify=False)
-        namespaced_record = pbclient.get_namespaced_keyvalue(NAMESPACE, KEY_CONFIG)
+        namespaced_record = pbclient.get_namespaced_keyvalue(namespace_value, KEY_CONFIG)
         driver_config = namespaced_record['value']
         return driver_config
 
     @staticmethod
-    def run_ansible_on_host(host, logger, driver_config):
+    def run_ansible_on_host(host, logger, driver_config, role):
```

Comment:
If we passed the playbook name instead of 'ntnormal' (maybe defaulting to 'notebook_playbook'), could we drop the condition on which playbook to run altogether? Also, for me it is not clear what 'ntnormal' means.

Reply:
Looks good now!